controllers/disruption_controller.go

```diff
@@ -431,6 +437,41 @@ func (r *DisruptionReconciler) cleanDisruption(instance *chaosv1beta1.Disruption
 	return cleaned, nil
 }
 
+func (r *DisruptionReconciler) handleOrphanedChaosPods(req ctrl.Request) error {
+	ls := make(map[string]string)
+
+	ls[chaostypes.DisruptionNameLabel] = req.Name
+	ls[chaostypes.DisruptionNamespaceLabel] = req.Namespace
+
+	chaosPods, err := r.getChaosPods(nil, ls)
+	if err != nil {
+		return err
+	}
+
+	for _, chaosPod := range chaosPods {
+		r.handleMetricSinkError(r.MetricsSink.MetricOrphanFound([]string{"name:" + req.Name, "namespace:" + req.Namespace}))
+
+		r.log.Warnw("orphaned chaos pod detected, will attempt to delete", "chaosPod", chaosPod.Name)
+		controllerutil.RemoveFinalizer(&chaosPod, chaostypes.ChaosPodFinalizer)
```

Comment:
What if the orphan pod fails to clean up the injected disruption? So far, it'll be deleted and removed without us knowing it. I feel like the lifecycle for those pods should (but probably can't?) be the same as for chaos pods with an attached disruption (deleting them, checking their status for a proper removal, etc).

Reply:
Hmm, great point. Let's reason about the situations in which we even have orphans:

1. Worst Case:
- The controller goes down. Someone then deletes a disruption resource and removes its finalizer. Without the controller to reconcile, the chaos pods are left uncleaned and orphaned.
2. Current Observed Case:
- We have a node failure disruption. After running, the node is gone, but chaos pods remain due to their finalizer. Somehow? `cleanDisruption` must be returning `isCleaned == true`, because the disruption itself cleans up.
3. "Other" Case:
- Overlap of (2), but `cleanDisruption` returns `isCleaned == true`, even though that's false, in which case the disruption has its finalizer removed.


It seems that theoretically we definitely can't just remove the finalizer from these orphans. In practice though, they've already been attempted to be deleted, and are stuck in Terminating, and just have the finalizer stuck for some reason. Maybe we should try deleting again before removing?